src/docfx/build/toc/BuildTableOfContents.cs

```diff
@@ -161,6 +113,14 @@ void ResolveItemMonikers(List<TableOfContentsItem> items)
                     }
 
                     return (link, display, buildItem);
+                },
+                (document) =>
+                {
+                    if (monikerMap != null && document != null && monikerMap.TryGetValue(document, out var monikers))
```

Comment:
feels like `MonikerMap` and `MonikerProvider` is the same thing, can we call `MonikerProvider.GetFileLevelMonikers` here instead? Can handle file --> moniker cache internally.

Reply:
we cannot get the file metadata here

---
In reply to: [291923815](https://github.com/dotnet/docfx/pull/4702#discussion_r291923815) [](ancestors = 291923815)

Reply:
should be able to get file metadata using `MetadataProvider`

Reply:
that method is just for document without yaml header, for now is  redirection and resource file.

---
In reply to: [291939668](https://github.com/dotnet/docfx/pull/4702#discussion_r291939668) [](ancestors = 291939668)